File: ts/04.删除排序链表中的重复元素.ts
function deleteDuplicates(head: ListNode | null): ListNode | null {
  const includedArr = [];
  const dummy = head;
  let prev: ListNode = null;
  while (head !== null) {
    const value = head.val;
    if (includedArr.includes(value)) {
      // 如果已经存在值
      prev.next = head.next;
      head = head.next;
    } else {
      // 如果不存在，将数值加入到数组中
      includedArr.push(value);
      prev = head;
      head = head.next;
    }
  }
  return dummy;
}

File: java/day02/04. 删除排序链表中的重复元素.java
/**
https://leetcode.cn/problems/remove-duplicates-from-sorted-list/description/
 */
/**
 * Definition for singly-linked list.
 * public class ListNode {
 * int val;
 * ListNode next;
 * ListNode() {}
 * ListNode(int val) { this.val = val; }
 * ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
  public ListNode deleteDuplicates(ListNode head) {
    if (head == null) {
      return null;
    }
    ListNode x = head;
    ListNode y = head.next;
    while (y != null) {
      if (x.val == y.val) {
        x.next = y.next;
        y = y.next;
      } else if (y != null) {
        x = x.next;
        y = y.next;
      }
    }
    return head;
  }
}